### PR TITLE
Add NewRelic Distributed Tracing support

### PR DESCRIPTION
**Pull request by iuri-gg**
This pr introduces client and server interceptors that can pass newrelic distributed tracing payload through `newrelic` header.

### CODE REVIEW

**Comment by iuri-gg** [comment]
@splittingred Let me know if there is anything you want to refactor/change. Otherwise, Im not planning to add anything else to this PR and it is ready for review/merge

File: lib/gruf/newrelic.rb [hunk 1]
```diff
@@ -24,6 +25,8 @@ module Gruf
   # Newrelic gruf module
   #
   module Newrelic
+    NEWRELIC_TRACE_HEADER = "newrelic".freeze
```
Comment (splittingred) [inline lib/gruf/newrelic.rb]: Can you do: 

```ruby
NEWRELIC_TRACE_HEADER = 'newrelic'
```

And add to the top of this file:

```
# frozen_string_literal: true
```

? Thanks!

File: lib/gruf/newrelic/client_interceptor.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    # New Relic transaction tracing for Gruf endpoints
+    #
+    class ClientInterceptor < ::Gruf::Interceptors::ClientInterceptor
+      def call(request_context:)
+        metadata = request_context.metadata
+        payload = ::NewRelic::Agent::DistributedTracing.create_distributed_trace_payload
+
+        if payload
+          metadata[NEWRELIC_TRACE_HEADER] = payload.http_safe
+        end
```
Comment (splittingred) [inline lib/gruf/newrelic/client_interceptor.rb]: Prefer guard conditionals over conditional blocks:

```
metadata[NEWRELIC_TRACE_HEADER] = payload.http_safe if payload
```

**Comment by iuri-gg** [comment]
would you like to have `frozen_string_literal` in all files? I can add to this PR. Also, I would recommend adding rubocop check to CI pipeline.

**Review by splittingred — APPROVED** [review]
Up to you. I'm happy to do that in follow-up work; this is good as-is to me :+1:

**Review by splittingred — APPROVED** [review]
Thanks!

**Comment by iuri-gg** [comment]
I added frozen_string_literal everywhere (`rubocop -a'). As for integrating rubocop to CI pipeline, I would rather not include it in this PR so it stays small and on topic

**Comment by splittingred** [comment]
@ocha This is great. I'm OOO the rest of the day, but can cut a release for this tomorrow. Thanks again for your contribution!

**Comment by iuri-gg** [comment]
Thank you and my pleasure

**Pull request merged**